dashboard/app/models/resource.rb

```diff
@@ -0,0 +1,15 @@
+# == Schema Information
+#
+# Table name: resources
+#
+#  id         :integer          not null, primary key
+#  name       :string(255)
```

Comment:
I think we want to include key as a column since that will be the immutable thing we use to keep track of resources across environments and for translation. We could then move name to be a property. In addition I think last week we decided to try to standardize on using title instead of name. Check out this pr comment for more context: https://github.com/code-dot-org/code-dot-org/pull/36772#discussion_r490678368.

Reply:
We should make sure key is unique and can not be null also

Reply:
hi @dmcavoy , what do you think about using `name` after all? sorry to waffle but I started using title and found I had to  keep explaining what it was. with `name`, the only downside is that a couple models use it as their unique "key", but everywhere else it kind of seems like the best option.

Reply:
@dmcavoy also when someone updates a resource, do you think we will ever want to search for other resources with the same name and offer to update those too? I think we will want to do this for url and key/slug, but unsure about name. if we do, then that would be a good reason for name to be its own column. otherwise it could go in properties.

Reply:
I'm fine with going back to name if thats what we want to settle on as a team. I don't have a strong opinion

Reply:
I guess I could see trying to search for resources with the same name as a way to try to help people know what to edit so maybe its worth it being a column

Reply:
^ that was what eventually convinced me to make `name` a column instead of a property. I went back and forth a bit so happy to change if it seems unnecessary.

I'll fix the key to be unique/non null.